src/Wolf.cpp

```diff
@@ -0,0 +1,466 @@
+  double correction = 0.0, dampenedCorr = 0.0, undampenedCorr = 0.0;
+  XYZ virComponents;
+
+  MoleculeKind& thisKind = mols.kinds[mols.kIndex[molIndex]];
+  uint atomSize = thisKind.NumAtoms();
+  uint start = mols.MolStart(molIndex);
+  double lambdaCoef = GetLambdaCoef(molIndex, box);
+
+  for (uint i = 0; i < atomSize; i++) {
+    if(particleHasNoCharge[start + i]) {
+      continue;
+    }
+    // This term only needs to be calculated once, if the molecules are rigid
+    // Or of the maximum radius of gyration < RCutCoulomb
+    // Otherwise, parts of the molecule may extend out of range of each other
+    // For now assume the latter.
+    for (uint j = i + 1; j < atomSize; j++) {
+      // Need to check for cutoff
+      if(currentAxes.InRcut(distSq, virComponents, currentCoords,
+                         start + i, start + j, box)){
+        // We eliminate the dampened rij term by choosing psi = 1
+        correction += -1.0 * thisKind.AtomCharge(i) * thisKind.AtomCharge(j) * wolfFactor1[box];
+      }
+    }
+  }
+
+  GOMC_EVENT_STOP(1, GomcProfileEvent::CORR_MOL);
+  return num::qqFact * correction * lambdaCoef * lambdaCoef;
+}
+
+
+//calculate reciprocal term in destination box for swap move
+double Wolf::SwapDestRecip(const cbmc::TrialMol &newMol,
+                              const uint box,
+                              const int molIndex)
+{
+  return 0.0;
+}
+
+
+//calculate reciprocal term in source box for swap move
+double Wolf::SwapSourceRecip(const cbmc::TrialMol &oldMol,
+                                const uint box, const int molIndex)
+{
+  return 0.0;
+}
+
+
+//calculate reciprocal term for inserting some molecules (kindA) in destination
+// box and removing a molecule (kindB) from destination box
+double Wolf::MolExchangeReciprocal(const std::vector<cbmc::TrialMol> &newMol,
+                                      const std::vector<cbmc::TrialMol> &oldMol,
+                                      const std::vector<uint> &molIndexNew,
+                                      const std::vector<uint> &molIndexold,
+                                      bool first_call)
+{
+  return 0.0;
+}
+
+
+//calculate self term after swap move
+double Wolf::SwapSelf(const cbmc::TrialMol& trialMol) const
+{
+  uint box = trialMol.GetBox();
+  if (box >= BOXES_WITH_U_NB)
+    return 0.0;
+
+  GOMC_EVENT_START(1, GomcProfileEvent::SELF_SWAP);
+  MoleculeKind const& thisKind = trialMol.GetKind();
+  uint atomSize = thisKind.NumAtoms();
+  double en_self = 0.0;
+
+  for (uint i = 0; i < atomSize; i++) {
+    en_self -= (thisKind.AtomCharge(i) * thisKind.AtomCharge(i));
+  }    
+
+  GOMC_EVENT_STOP(1, GomcProfileEvent::SELF_SWAP);
+  // M_2_SQRTPI is 2/sqrt(PI), so need to multiply by 0.5 to get sqrt(PI)
+  // Vlugt
+  //self *= ((ff.wolfAlpha[box] * M_2_SQRTPI * 0.5) +  0.5 * ff.wolfFactor1[box]);
+  // we eliminate the alpha/root(pi) using Wolf,mod from Gross et al
+  return (en_self *= -1.0 * ff.wolfFactor1[box] * num::qqFact);
+}
```

Comment:
I dont think you should multiply bu `-1.0` again, since you have `en_self -= (thisKind.AtomCharge(i) * thisKind.AtomCharge(i));`

Reply:
Thanks, I am looking at the signs of all the Self and Correction terms.  They are consistent in Ewald, so I'll make them consistent in Wolf.